docs/source/checks/tabular/model_evaluation/plot_weak_segments_performance.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+"""
+Weak Segments Performance
+********************************
+
+This notebooks provides an overview for using and understanding the weak segment performance check.
+
+**Structure:**
+
+* `What is the purpose of the check? <#what-is-the-purpose-of-the-check>`__
+* `Generate data & model <#generate-data-model>`__
+* `Run the check <#run-the-check>`__
+* `Define a condition <#define-a-condition>`__
+
+What is the purpose of the check?
+=================================
+The check is designed to help you easily identify the model's weakest segments in the data provided. In addition,
+it is possible to provide a sub list of features for the check thus limiting the search to interesting sub-spaces.
+
+How Deepchecks detects weak segments
+------------------------------------
+The check contains several steps:
+
+#. We calculate loss for each sample in the dataset using the provided model via either log-loss or MSE.
+
+#. Select a subset of features of the weak segment search. This is done by selecting the features with the highest feature importance to the model provided.
+
+#. We train multiple simple tree based models, each one is trained based on only two features to predict the per sample error calculated before.
+
+#. We convert each of the leafs in each of the trees into a segment and analyze the segment's performance.
+
+#. For the model's weakest segments detected we calculate bins for the remaining of the data and calculate the model's
+"""
+#%%
+# Generate data & model
+# =====================
+
+from deepchecks.tabular.datasets.classification.phishing import (
+    load_data, load_fitted_model)
+
+train_dataset, test_dataset = load_data()
+model = load_fitted_model()
+
+#%%
+# Run the check
+# =============
+# The check has several key parameters that can define the behavior of the check and especially its output.
+#
+# columns / ignore_columns: Controls which columns should be searched for weak segments. By default,
+# a heuristic is used to determine which columns to use based on their feature importance.
+#
+# alternative_scorer: Determines the metric to be used as the performance measurement of the model on different
```

Comment:
state something about sklearn scorers...?

Reply:
Not at the moment, would add a link to metrics guide when it is complete